File: Code_CS/C18_AppLogic/App_Code/TextLoggerModule.cs
using System;
using System.Web;

/// <summary>
/// Summary description for TextLoggerModule
/// </summary>
public class TextLoggerModule : IHttpModule
{
    #region IHttpModule Members

    public void Dispose()
    {
        throw new NotImplementedException();
    }

    public void Init(HttpApplication context)
    {
        context.PreRequestHandlerExecute += new EventHandler(context_PreRequestHandlerExecute);
    }

    void context_PreRequestHandlerExecute(object sender, EventArgs e)
    {
        GlobalMembers.StaticWriteFile(HttpContext.Current.Request.RawUrl);
    }

    #endregion
}

File: Code_VB/C18_AppLogic/App_Code/TextLoggerModule.vb
Imports System
Imports System.Web

''' <summary> 
''' Summary description for TextLoggerModule 
''' </summary> 
Public Class TextLoggerModule
    Implements IHttpModule

#Region "IHttpModule Members"

    Public Sub Dispose() Implements System.Web.IHttpModule.Dispose
        Throw New NotImplementedException()
    End Sub

    Public Sub Init(ByVal context As HttpApplication) Implements IHttpModule.Init
        AddHandler context.PreRequestHandlerExecute, AddressOf context_PreRequestHandlerExecute
    End Sub

    Private Sub context_PreRequestHandlerExecute(ByVal sender As Object, ByVal e As EventArgs)
        GlobalMembers.StaticWriteFile(HttpContext.Current.Request.RawUrl)
    End Sub

#End Region

End Class

